libs/ml/include/ml/graph.hpp

```diff
@@ -96,6 +105,32 @@ class Graph
   std::vector<TrainablePtrType>                trainable_;
 };
 
+template <typename ArrayType>
+void Graph<ArrayType>::SetRegularisation(fetch::ml::details::RegularisationType regularisation,
+                                         DataType                               regularisation_rate)
+{
+  for (auto &t : trainable_)
+  {
+    t->SetRegularisation(regularisation, regularisation_rate);
+  }
+}
+
+template <typename ArrayType>
+void Graph<ArrayType>::SetRegularisation(std::string                            node_name,
+                                         fetch::ml::details::RegularisationType regularisation,
+                                         DataType                               regularisation_rate)
+{
+  if (nodes_[node_name])
+  {
+    trainable_[node_name]->SetRegularisation(regularisation, regularisation_rate);
+  }
+  else
+  {
+    throw std::runtime_error("Cannot change regularisation parameters for node [" + node_name +
```

Comment:
how about we change the design so that SetRegularisation returns a success bool instead of throwing? I think in general anything that get used in the VM will need to return a success bool so that the VM can handle the error in whatever way is most suitable.